contracts/Solver.sol

```diff
@@ -5,17 +5,21 @@ import './TaskGiver.sol';
 
 contract Solver is AccountManager {
 
+	bytes32 random = "12345678";
 	mapping(uint => mapping(address => address)) challenges;
 
 	event SendSolution(address solver, address taskGiver, uint id, bytes32 solution, uint minDeposit, bytes32 taskData);
 
 	function sendBid(address origin, uint id, uint minDeposit, address addr) {
 		require(balances[addr] >= minDeposit);
 		require(TaskGiver(origin).receiveBid(id, addr));
+		random = sha3(random, now);
 	}
 
 	function solveTask(address solver, address taskGiver, bytes32 taskData, uint id, uint minDeposit) returns (bool) {
 		bytes32 solution = 0x0;
+		//add send correct or incorrect solution here
+		//sha3(random, block.blockhash)
```

Comment:
I think the blockhash has more randomness than timestamp, it can be called using `block.blockhash(block.number-1)`. I don't know if it works for testing.

Reply:
Interesting, I'll try it out.